### PR TITLE
Seeking more incremental feedback on Autocorr

### PR DESCRIPTION
**Pull request by canyon289**
Per @ColCarroll's comments added an example of an autocorrelation plot that looks poor.

Have not included @ahartikainen's recommendation of interactivity or parts of a plot 

Goal for this doc is to iterate a lot to get it to a high standard, then use this as the benchmark for the docs for the rest of the plots

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 arviz
+jupyter-lab
```
Comment (aloctavodia) [inline requirements.txt]: Why is jupyter-lab a requirement and not just jupyter?

Comment (canyon289) [reply requirements.txt]: I recently went to a talk from one of the jupyter core devs and was sufficiently convinced jupyter-lab should be favored over legacy jupyter application these days.

File: plots/AutoCorrelation.ipynb [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+   "source": [
+    "## Autocorrelation deriviation\n",
+    "Autocorrelation at each lag is defined as"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "$$\\large \\rho_{lag} = \\frac{\\Sigma_{i}^{N-lag} (\\theta_{i} - \\overline{\\theta})(\\theta_{i + lag} - \\overline{\\theta})}\n",
```
[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: To follow up on the slack comments, sounds good

File: plots/AutoCorrelation.ipynb [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
...e4gPDxckmQymezWW8oty9VV8+bN5ePjo5KSEpWVlVl36B3VPwAAAOAMGvwxoDExMZKkrKysKnX5+fkymUwKCgpSSEjIdfVjMplUUlIib29vm/TkqP4BAAAAZ9DgASA6OlqdO3fWrl27lJmZaS2vqKjQnDlzJEmJiYk21+CXl5crNzdXubm5Nm3l5+fr5MmTVfo4c+aMpk2bJkm6//775e7+fyc26tI/AAAAcKNyqXTAN2Dl5OQoKSlJly9f1sCBA9W+fXtt27ZN+/btU1RUlJYuXWpzj8HRo0etX9p14MABa/natWuVmpqq3r17KzQ0VH5+fjp+/Li2bt2qs2fPqkePHnr33XfVsmXL6+r/SpcvX+Y6Pdjg2k3Yw7yAPcwL2MO8gD1N6h6A6xUZGakPPvhACxYsUHZ2ts6fP6+goCCNHz9eY8aMqfENxt27d9fgwYO1b98+7d+/X2azWd7e3uratavuu+8+JSQk2G2rvvoHAAAAnJ1DzgA4O84A4EocuYE9zAvYw7yAPcwL2OOoMwANfg8AAAAAgKaDAAAAAAAYCAEAAAAAMBACAAAAAGAgBAAAAADAQAgAAAAAgIEQAAAAAAADIQAAAAAABkIAAAAAAAyEAAAAAAAYCAEAAAAAMBACAAAAAGAgBAAAAADAQAgAAAAAgIEQAAAAAAADIQAAAAAABkIAAAAAAAyEAAAAAAAYCAEAAAAAMBACAAAAAGAgBAAAAADAQAgAAAAAgIEQAAAAAAADIQAAAAAABkIAAAAAAAyEAAAAAAAYiLujOjp48KDS0tK0c+dOmc1mdejQQffff7/GjBkjT0/PGrWxY8cOjRo16prLLV++XL169bL+HhcXp4KCArvLtmnTRl9++WXNNgIAAABwcg4JADk5OUpKStLly5c1cOBAtWvXTtu3b1dGRoa2bdumpUuXysPD45rtBAUFacKECXbrjh49qnXr1snf3189e/a0u4y9db28vGq3MQAAAIATa/AA8OOPP2ratGkqKyvTwoULFR8fL0mqqKjQk08+qY8//lhLlizRmDFjrtlWcHCwUlJS7Na9/PLLkqThw4dXGyaqWxcAAAAwiga/B2Dnzp3Kzc1V7969rTv/kuTq6qqpU6dKklauXKnKyso693Hx4kWtX79eLi4uSkhIuO4xAwAAADeqBj8DsH37dklSv379qtSFhIQoLCxMJ
```
[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: We could, the default figures were small so I wanted to make them a bit bigger. For my understanding what undesirable side effect does this have?

[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: Above you say `High autocorrelation can indicate lack of convergence` but here say it `does not necessary imply lack of convergence`. Does imply and indicate mean two separate things?

[the remark this answers is not in the document]
Comment (aloctavodia) [reply plots/AutoCorrelation.ipynb]: They should be bigger if the style is set in the second cell. No problem if you want to pass the figsize values.

[the remark this answers is not in the document]
Comment (aloctavodia) [reply plots/AutoCorrelation.ipynb]: I would say "can indicate" is about a possibility while "imply" means with total certainty.

File: requirements.txt (hunk 1 above)
Comment (aloctavodia) [reply requirements.txt]: right I have been hearing the same, probably a good idea to help with the adoption of jupyter-lab

File: plots/AutoCorrelation.ipynb [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+   "metadata": {},
+   "source": [
+    "# Autocorrelation Diagnostic"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Autocorrelation plots show the degree a time series signal is correlated with itself.  As a diagnostic for Markov Monte Carlo Chains they provides an estimate the indepedence of each draw from previous draws in the same chain. This is important since draws from a mcmc chain that are highly correlated could indicate issues with chain convergence."
```
[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: Changed to `Autocorrelation plots show an estimate of the independence of a time series signal with itself.`. Autocorrelation is used outside of MCMC chains in other fields so want to mention that

File: plots/AutoCorrelation.ipynb [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([[<matplotlib.axes._subplots.AxesSubplot object at 0x1c18753080>,\n",
```
[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: This suppresses both the `repr` and the plot though. I changed to `ax = az.plot_autocorr(data, var_names="mu")` which hides the `repr` but keeps the plots

File: plots/AutoCorrelation.ipynb [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+    }
+   ],
+   "source": [
+    "import arviz as az\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "\n",
+    "az.style.use('arviz-darkgrid')\n",
+    "data = az.load_arviz_data('centered_eight')\n",
+    "az.plot_autocorr(data, var_names=\"mu\")"
```
[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: This is correct in the notebook? I think jupyter is adding in the backslashes as quote escapes

File: plots/AutoCorrelation.ipynb [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "fig, ax = plt.subplots(figsize=(12,6))\n",
+    "ax.vlines(x=np.arange(1, 100),\n",
+    "                  ymin=0, ymax=auto_corr,\n",
+    "                  lw=1)\n",
+    "fig.suptitle(\"Autocorrelation of Mu Chain 0\")"
```
[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: Confused on this comment. I don't see what the difference is?

[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: Oh I see, the semicolon. I'll make the same variable assignment fix

File: plots/AutoCorrelation.ipynb (hunk 3 above)
[the remark this answers is not in the document]
Comment (canyon289) [reply plots/AutoCorrelation.ipynb]: Your comment is pasted in word for word in the docs :)

File: plots/AutoCorrelation.ipynb (hunk 5 above)
[the remark this answers is not in the document]
Comment (aloctavodia) [reply plots/AutoCorrelation.ipynb]: You are right, the last line should not be the one generating the plot!

File: plots/AutoCorrelation.ipynb (hunk 7 above)
[the remark this answers is not in the document]
Comment (aloctavodia) [reply plots/AutoCorrelation.ipynb]: here you can use the semicolon, as this is not the line generating the plot. No need to assign a variable.

**Comment by canyon289** [comment]
Marked as Alpha and added todo note to add neffective.
Resolved feedback for excess code output by semicolon or assigning vars were appropriate
Merging due to the above

**Pull request merged**